eth/watchers/unbondingwatcher.go

```diff
@@ -0,0 +1,162 @@
+package watchers
+
+import (
+	"fmt"
+	"math/big"
+
+	ethereum "github.com/ethereum/go-ethereum"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/golang/glog"
+	"github.com/livepeer/go-livepeer/eth/blockwatch"
+	"github.com/livepeer/go-livepeer/eth/contracts"
+)
+
+type unbondingLockStore interface {
+	InsertUnbondingLock(id *big.Int, delegator ethcommon.Address, amount, withdrawRound *big.Int) error
+	DeleteUnbondingLock(id *big.Int, delegator ethcommon.Address) error
+	UseUnbondingLock(id *big.Int, delegator ethcommon.Address, usedBlock *big.Int) error
+}
+
+// UnbondingWatcher watches for on-chain events to update the state of an unbonding lock store
+type UnbondingWatcher struct {
+	addr  ethcommon.Address // Watching for on-chain events pertaining to this address
+	bw    blockWatcher
+	store unbondingLockStore
+	dec   *EventDecoder
+	sub   ethereum.Subscription
+
+	quit chan struct{}
+}
+
+// NewUnbondingWatcher creates an UnbondingWatcher instance
+func NewUnbondingWatcher(addr ethcommon.Address, bondingManagerAddr ethcommon.Address, bw blockWatcher, store unbondingLockStore) (*UnbondingWatcher, error) {
+	dec, err := NewEventDecoder(bondingManagerAddr, contracts.BondingManagerABI)
+	if err != nil {
+		return nil, err
+	}
+
+	return &UnbondingWatcher{
+		addr:  addr,
+		bw:    bw,
+		store: store,
+		dec:   dec,
+		quit:  make(chan struct{}),
+	}, nil
+}
+
+// Start kicks off a loop that handles events from a block subscription
+func (w *UnbondingWatcher) Start() {
+	blockEvents := make(chan []*blockwatch.Event, 10)
+	w.sub = w.bw.Subscribe(blockEvents)
+
+	for {
+		select {
+		case <-w.quit:
+			w.sub.Unsubscribe()
```

Comment:
I have a slight preference for deferring the unsubscribe right after subscribing and simply returning from `Start()` when we close the `quit` channel so that we make sure we unsubscribe for all cases when the `Start()` function returns. WDYT?

Reply:
Fixed in 3ba3b20